lib/puppet/application/apply.rb

```diff
@@ -247,7 +248,9 @@ def main
 
         catalog.retrieval_duration = Time.now - starttime
 
-        if options[:write_catalog_summary]
+        # option parsing converts dashes to underscores, and we want
+        # to prefer the option for backwards compatibility
+        if options[:write_catalog_summary] || Puppet[:write_catalog_summary]
```

Comment:
Assuming that the user does not pass `write_catalog_summary` via CLI or config, won't `Puppet[:write_catalog_summary]` be `true` by default and always write the files? The PR description suggests that it shouldn't.

Reply:
doh, let me fix that

Reply:
Ah it actually works as expected because the `apply` application overrides the default in https://github.com/puppetlabs/puppet/pull/8753/commits/df9d93626c8cb37ab4e2c34ba64b53aec1aba238#diff-375b738c02ca2ff08e306eb14b5287d114ba835a5d3430c822f87f995e443a05R172 But I'll a comment.